pkg/controller/osimagestream/urls_source.go

```diff
@@ -0,0 +1,120 @@
+package osimagestream
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/api/machineconfiguration/v1alpha1"
+	"github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/imageutils"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+)
+
+// OSImageTuple represents a pair of container images for OS and extensions.
+type OSImageTuple struct {
+	OSImage           string
```

Comment:
thought (non-blocking): We may eventually want to map `common.OSImageURLConfig` to use this type instead of what we're currently using.